library/lib/components/button/ods_text_button.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:flutter/material.dart';
+
+/// ODS Design Button.
+///
+/// Displays a customizable button with an optional icon and a title.
+/// The button can expand to full screen width if specified.
+class OdsTextButton extends StatefulWidget {
+  /// Creates an ODS Text Button.
+  ///
+  /// * [title] - Text displayed in the button.
+  /// * [fullScreenWidth] - Specifies whether the button should expand to full screen width.
+  /// * [icon] - Widget of the icon.
+  /// * [onPressed] - The action to be executed when the button is pressed.
+  const OdsTextButton({
+    Key? key,
+    required this.title,
+    required this.fullScreenWidth,
+    this.icon,
+    this.onPressed,
+  }) : super(key: key);
+
+  /// The button's title displayed inside the button.
+  final String title;
+
+  /// Specifies whether the button should expand to full screen width.
+  final bool fullScreenWidth;
+
+  /// The optional button's icon.
+  final Widget? icon;
+
+  /// The action to be executed when the button is pressed.
+  final Function()? onPressed;
+
+  @override
+  State<OdsTextButton> createState() => _OdsTextButtonState();
+}
+
+class _OdsTextButtonState extends State<OdsTextButton> {
```

Comment:
is ti possible to mutualize the code for all buttons ?

Reply:
I have not found a solution to mutualize the code for all buttons